lib/chef/resource.rb

```diff
@@ -1169,7 +1236,9 @@ def validate_action(action)
     end
 
     def provider_for_action(action)
-      provider = Chef::ProviderResolver.new(node, self, action).resolve.new(self, run_context)
+      provider_class = self.class.action_provider_class
+      provider_class ||= Chef::ProviderResolver.new(node, self, action).resolve
```

Comment:
Actually, since we're setting `def provider` we don't really need to do that (same behavior, no actual resolution even though it calls the Resolver), so I'm going to remove that line.  Thanks!